### PR TITLE
Added the email domain as a data source

### PR DESCRIPTION
**Pull request by carnellj**

### CODE REVIEW

**Comment by joefruland** [comment]
Run `go generate` in the repo root directory to add this to the documenation.

File: genesyscloud/data_source_genesyscloud_routing_email_domain_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+)
+
+func TestAccDataSourceRoutingEmailDomain(t *testing.T) {
+	var (
+		emailDomainResourceId   = "email_domain_test"
+		emailDomainId  			= "jcc4234234.thoughtx.com"
+		emailDataResourceId 	= "email_domain_data"
+	)
+
+	fmt.Printf(generateRoutingEmailDomainResource(
```
Comment (joefruland) [inline genesyscloud/data_source_genesyscloud_routing_email_domain_test.go:16]: This printf can be removed

File: genesyscloud/data_source_genesyscloud_routing_email_domain_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"testing"
+)
+
+func TestAccDataSourceRoutingEmailDomain(t *testing.T) {
+	var (
+		emailDomainResourceId   = "email_domain_test"
+		emailDomainId  			= "jcc4234234.thoughtx.com"
```
Comment (joefruland) [inline genesyscloud/data_source_genesyscloud_routing_email_domain_test.go:12]: Uneven spacing here.

File: genesyscloud/provider.go [hunk 3]
```diff
@@ -104,6 +104,7 @@ func New(version string) func() *schema.Provider {
 				"genesyscloud_flow":             dataSourceFlow(),
 				"genesyscloud_routing_language": dataSourceRoutingLanguage(),
 				"genesyscloud_routing_skill":    dataSourceRoutingSkill(),
+				"genesyscloud_routing_email_domain":             dataSourceRoutingEmailDomain(),
```
Comment (joefruland) [inline genesyscloud/provider.go:107]: Uneven spacing here.

File: genesyscloud/data_source_genesyscloud_routing_email_domain_test.go (hunk 1 above)
Comment (carnellj) [reply genesyscloud/data_source_genesyscloud_routing_email_domain_test.go:16]: Doh! Missed that.  Put that in there while debugging

File: genesyscloud/data_source_genesyscloud_routing_email_domain_test.go (hunk 2 above)
Comment (carnellj) [reply genesyscloud/data_source_genesyscloud_routing_email_domain_test.go:12]: Fixed

File: genesyscloud/provider.go (hunk 3 above)
Comment (carnellj) [reply genesyscloud/provider.go:107]: Fixed

File: genesyscloud/data_source_genesyscloud_routing_email_domain_test.go (hunk 1 above)
Comment (carnellj) [reply genesyscloud/data_source_genesyscloud_routing_email_domain_test.go:16]: Fixed

File: genesyscloud/resource_genesyscloud_routing_email_domain_test.go [hunk 4]
```diff
@@ -26,11 +26,11 @@ func TestAccResourceRoutingEmailDomainSub(t *testing.T) {
 				Config: generateRoutingEmailDomainResource(
 					domainRes,
 					domainId,
-					trueValue, // Subdomain
+					trueValue, // Subdomain clear
 					nullValue,
 				),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("genesyscloud_routing_email_domain."+domainRes, "domain_id", domainId),
+					//resource.TestCheckResourceAttr("genesyscloud_routing_email_domain."+domainRes, "domain_id", domainId ),
```
Comment (joefruland) [inline genesyscloud/resource_genesyscloud_routing_email_domain_test.go:33]: The change to fix subdomains in other regions is in the latest release. You should be able to uncomment this line.

**Pull request merged**